sdk/protocol2.go

```diff
@@ -132,6 +132,19 @@ func (d *EntityData) NewMetricSet(eventType string) metric.MetricSet {
 	return ms
 }
 
+// Event method creates or retrieves an already created Event.
+func (d *EntityData) Event(summary, category string) *Event {
+	for _, e := range d.Events {
+		if e.Summary == summary && e.Category == category {
+			return e
+		}
+	}
+	ev := NewEvent(summary, category)
+	d.Events = append(d.Events, &ev)
```

Comment:
Have in mind that we collect event in a non thread safe way. What I mean is that if we have several goroutines calling this method at the same time will cause a race resulting in a wrong state.

There is a TODO task for our team to fix that in a general way but maybe we could do it now here.

Reply:
Good point @smoya. However, I would prefer keeping the scope of this PR, and we could address that in a different PR.

Reply:
Yes, I agree with @areina, because I won't have time to finish it and test it.